apply_change_to_project: Handle remove configuration list change.

byohay opened the pull request:
Also, refactor applying added changes for symmetry with removed changes.

ashdnazg commented on lib/kintsugi/apply_change_to_project.rb:
```diff
@@ -210,23 +210,30 @@ def apply_change_to_component(parent_component, change_name, change, parent_chan
+      elsif change[:removed].is_a?(Array)
+        (change[:removed]).each do |removed_change|
+          child =
+            child_component_of_object_list(component, removed_change, removed_change["displayName"])
```
this function expects component to be non-nil

ashdnazg commented on lib/kintsugi/apply_change_to_project.rb:
```diff
@@ -210,23 +210,30 @@ def apply_change_to_component(parent_component, change_name, change, parent_chan
-        add_child_to_component(is_object_list ? parent_component : component, added_change,
-                               change_path)
+      if change[:added].is_a?(Hash)
+        add_child_to_component(parent_component, change[:added], change_path)
```
same

ashdnazg reviewed (changes requested):
F&S

byohay commented:
@ashdnazg I added guards around places that need `component`. I also added this dependency in change[:added], as it always needs the parent component.

ashdnazg reviewed (approved):
I think it's ok,
Note that in the commit message you still mention symmetry, but that's no longer the case.
Also we need to remember to fail `added` if it will override a different existing value. Probably in a different PR.
